public/app/features/alerting/unified/components/rule-editor/LabelsField.tsx

```diff
@@ -1,17 +1,65 @@
 import { css, cx } from '@emotion/css';
-import React, { FC, useMemo } from 'react';
-import { FieldArrayWithId, useFieldArray, useFormContext } from 'react-hook-form';
+import { flattenDeep, compact, uniq } from 'lodash';
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+import { useFieldArray, useFormContext } from 'react-hook-form';
 
 import { GrafanaTheme, SelectableValue } from '@grafana/data';
-import { Button, Field, InlineLabel, Label, useStyles } from '@grafana/ui';
+import { Stack } from '@grafana/experimental';
+import { Button, Field, InlineLabel, Label, useStyles, Tooltip, Icon } from '@grafana/ui';
+import { useDispatch } from 'app/types';
+import { RulerRuleGroupDTO } from 'app/types/unified-alerting-dto';
 
+import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
+import { fetchRulerRulesIfNotFetchedYet } from '../../state/actions';
 import { RuleFormValues } from '../../types/rule-form';
-import AlertLabelDropdown, { AlertLabelDropdownProps } from '../AlertLabelDropdown';
+import { GRAFANA_RULES_SOURCE_NAME } from '../../utils/datasource';
+import AlertLabelDropdown from '../AlertLabelDropdown';
 
 interface Props {
   className?: string;
 }
 
+const useGetCustomLabels = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchRulerRulesIfNotFetchedYet(GRAFANA_RULES_SOURCE_NAME));
+  }, [dispatch]);
+
+  const rulerRuleRequests = useUnifiedAlertingSelector((state) => state.rulerRules);
+
+  const rulerRequest = rulerRuleRequests[GRAFANA_RULES_SOURCE_NAME];
```

Comment:
What happens for Mimir/Cortex/Loki data sources? If we display for them labels from Grafana then users might be confused

Reply:
I think dataSourceName should be passed as a prop in LabelsField

Reply:
Do you think I should fetch rules for all the different datasources?

Reply:
I made some changes to only fetch the rules and display the labels for the current `dataSourceName` which is taken from the form context.

![2022-11-01 17 26 14](https://user-images.githubusercontent.com/6271380/199334352-943430b1-6f2d-4b17-970f-843a34ef678f.gif)

Reply:
If we want to have a re-usable `LabelField` we should not depend on the form that this `LabelField` is used. In contact points view, for example, we are using a different form.
So, I think that this `LabelField` component should receive `dataSource` as a prop, and the parent component should be the responsible for passing the correct value.